datastream-server-restli/src/test/java/com/linkedin/datastream/server/TestCoordinator.java

```diff
@@ -510,6 +510,29 @@ public void testCoordinationWithBroadcastStrategy() throws Exception {
     instance1.getDatastreamCache().getZkclient().close();
   }
 
+  @Test
+  public void testHandleAssignmentChangeTransientFailure() throws Exception {
+    String testCluster = "testCoordinationSmoke";
+    String testConnectorType = "testConnectorType";
+    String datastreamName1 = "datastream1";
+
+    DummyTransportProviderAdminFactory dummyTransportProviderAdminFactory = new DummyTransportProviderAdminFactory(
+        false, true);
+    Coordinator instance1 = createCoordinator(_zkConnectionString, testCluster, new Properties(),
+        dummyTransportProviderAdminFactory);
+    TestHookConnector connector1 = new TestHookConnector("connector1", testConnectorType);
+    instance1.addConnector(testConnectorType, connector1, new BroadcastStrategy(Optional.empty()), false,
+        new SourceBasedDeduper(), null);
+    instance1.start();
+
+    ZkClient zkClient = new ZkClient(_zkConnectionString);
+    DatastreamTestUtils.createAndStoreDatastreams(zkClient, testCluster, testConnectorType, datastreamName1);
+    //verify the assignment
+    assertConnectorAssignment(connector1, WAIT_TIMEOUT_MS, datastreamName1);
+    String instance1Path = KeyBuilder.instanceAssignments(testCluster, instance1.getInstanceName());
+    Assert.assertNotEquals(zkClient.getChildren(instance1Path).size(), 0);
```

Comment:
Cleanup the instantiated objects like zkclient, instance1. Some of these tests run in parallel and these open object handlers create issues.

Reply:
Done